applications/FluidDynamicsApplication/custom_elements/stokes_3D_twofluid.cpp

```diff
@@ -25,981 +25,996 @@ void Stokes3DTwoFluid::ComputeGaussPointLHSContribution(BoundedMatrix<double,16,
+lhs(13,2)=-clhs12*clhs345 - clhs12*clhs346 - clhs12*clhs348 + clhs344 + clhs349 + clhs350 + clhs351;
+lhs(13,3)=-clhs374;
+lhs(13,4)=-clhs12*clhs433 - clhs12*clhs434 - clhs12*clhs435 + clhs432 + clhs436 + clhs437 + clhs438;
+lhs(13,5)=-clhs12*clhs490 - clhs12*clhs491 - clhs12*clhs492 + clhs493 + clhs494 + clhs495 + clhs497;
+lhs(13,6)=-clhs12*clhs543 - clhs12*clhs544 - clhs12*clhs545 + clhs542 + clhs546 + clhs547 + clhs548;
+lhs(13,7)=-clhs563;
+lhs(13,8)=-clhs12*clhs597 - clhs12*clhs598 - clhs12*clhs599 + clhs596 + clhs600 + clhs601 + clhs602;
+lhs(13,9)=-clhs12*clhs631 - clhs12*clhs632 - clhs12*clhs633 + clhs634 + clhs635 + clhs636 + clhs638;
+lhs(13,10)=-clhs12*clhs661 - clhs12*clhs662 - clhs12*clhs663 + clhs660 + clhs664 + clhs665 + clhs666;
+lhs(13,11)=-clhs677;
+lhs(13,12)=-clhs12*clhs687 - clhs12*clhs688 - clhs12*clhs689 + clhs686 + clhs690 + clhs691 + clhs692;
+lhs(13,13)=-clhs12*clhs702 - clhs12*clhs703 - clhs12*clhs704 - clhs3*clhs701 + clhs680 + clhs702 + clhs703 + clhs704;
+lhs(13,14)=-clhs12*clhs709 - clhs12*clhs710 - clhs12*clhs711 + clhs705 + clhs706 + clhs707 + clhs708;
+lhs(13,15)=-clhs712;
+lhs(14,0)=-clhs12*clhs153 - clhs12*clhs156 - clhs12*clhs159 + clhs151 + clhs160 + clhs161 + clhs162;
+lhs(14,1)=-clhs12*clhs267 - clhs12*clhs269 - clhs12*clhs272 + clhs266 + clhs273 + clhs274 + clhs275;
+lhs(14,2)=-clhs12*clhs352 - clhs12*clhs353 - clhs12*clhs355 + clhs356 + clhs357 + clhs358 + clhs360;
+lhs(14,3)=-clhs375;
+lhs(14,4)=-clhs12*clhs440 - clhs12*clhs441 - clhs12*clhs442 + clhs439 + clhs443 + clhs444 + clhs445;
+lhs(14,5)=-clhs12*clhs499 - clhs12*clhs500 - clhs12*clhs501 + clhs498 + clhs502 + clhs503 + clhs504;
+lhs(14,6)=-clhs12*clhs549 - clhs12*clhs550 - clhs12*clhs551 + clhs552 + clhs553 + clhs554 + clhs556;
+lhs(14,7)=-clhs564;
+lhs(14,8)=-clhs12*clhs604 - clhs12*clhs605 - clhs12*clhs606 + clhs603 + clhs607 + clhs608 + clhs609;
+lhs(14,9)=-clhs12*clhs640 - clhs12*clhs641 - clhs12*clhs642 + clhs639 + clhs643 + clhs644 + clhs645;
+lhs(14,10)=-clhs12*clhs667 - clhs12*clhs668 - clhs12*clhs669 + clhs670 + clhs671 + clhs672 + clhs674;
+lhs(14,11)=-clhs678;
+lhs(14,12)=-clhs12*clhs694 - clhs12*clhs695 - clhs12*clhs696 + clhs693 + clhs697 + clhs698 + clhs699;
+lhs(14,13)=-clhs12*clhs706 - clhs12*clhs707 - clhs12*clhs708 + clhs705 + clhs709 + clhs710 + clhs711;
+lhs(14,14)=-clhs12*clhs714 - clhs12*clhs715 - clhs12*clhs716 - clhs3*clhs713 + clhs680 + clhs714 + clhs715 + clhs716;
+lhs(14,15)=-clhs717;
+lhs(15,0)=clhs163 + clhs362*clhs373;
+lhs(15,1)=clhs276 + clhs362*clhs374;
+lhs(15,2)=clhs361 + clhs362*clhs375;
+lhs(15,3)=clhs376;
+lhs(15,4)=clhs362*clhs562 + clhs446;
+lhs(15,5)=clhs362*clhs563 + clhs505;
+lhs(15,6)=clhs362*clhs564 + clhs557;
+lhs(15,7)=clhs565;
+lhs(15,8)=clhs362*clhs676 + clhs610;
+lhs(15,9)=clhs362*clhs677 + clhs646;
+lhs(15,10)=clhs362*clhs678 + clhs675;
+lhs(15,11)=clhs679;
+lhs(15,12)=clhs363*clhs700;
+lhs(15,13)=clhs363*clhs712;
+lhs(15,14)=clhs363*clhs717;
+lhs(15,15)=clhs364*(clhs681 + clhs701 + clhs713);
```

Comment:
note that there is another opportunity for opimization if you do something like

lhs(.,.) += weight*(...)

this was explored in some other PR

Reply:
Thanks Riccardo, i'll give a try.

Reply:
Weight is not present in element_data, so it requires to add it and compute it. As it's a bigger development than just modifying template, i prefer doing it in next PR.